src/us/kbase/workspace/test/database/provenance/EventDateTest.java

```diff
@@ -122,27 +121,43 @@ public void buildAndTrimEvent() throws Exception {
 	@Test
 	public void buildEventDateFailEvent() throws Exception {
 		buildEventDateFailWithError(
-			DATE_STRING, "kookaburra", new IllegalArgumentException(
-				"Invalid event: kookaburra"
-			)
-		);
+				DATE_STRING, "kookaburra",
+				"Invalid event: kookaburra");
 	}
 
 	@Test
 	public void buildEventDateFailNullEvent() throws Exception {
-		buildEventDateFailWithError(DATE_STRING, (Event) null, new NullPointerException("event cannot be null"));
+		buildEventDateFailWithError(DATE_STRING, (Event) null, "event cannot be null");
 	}
 
-
 	@Test
-	public void buildFailNullOrWhitespaceEventDate() throws Exception {
+	public void buildFailNullOrWhitespaceDate() throws Exception {
 		for (String nullOrWs : WHITESPACE_STRINGS_WITH_NULL) {
+			buildEventDateFailWithError(nullOrWs, EVENT_STRING, DATE_NO_NULL_WS);
+		}
+	}
 
-			buildEventDateFailWithError(nullOrWs, EVENT_STRING, new IllegalArgumentException("date cannot be null or whitespace only"));
+	@Test
+	public void buildFailNullOrWhitespaceEvent() throws Exception {
+		for (String nullOrWs : WHITESPACE_STRINGS_WITH_NULL) {
+			buildEventDateFailWithError(DATE_STRING, nullOrWs, EVENT_NO_NULL_WS);
+		}
+	}
 
-			buildEventDateFailWithError(DATE_STRING, nullOrWs, new IllegalArgumentException("event cannot be null or whitespace only"));
+	@Test
+	public void buildFailNullOrWhitespaceEventDate() throws Exception {
+		for (String nullOrWs : WHITESPACE_STRINGS_WITH_NULL) {
+			buildEventDateFailWithError(nullOrWs, nullOrWs, EVENT_NO_NULL_WS + "\n" + DATE_NO_NULL_WS);
+		}
+	}
 
-			buildEventDateFailWithError(nullOrWs, nullOrWs, new IllegalArgumentException("event cannot be null or whitespace only"));
+	@Test
+	public void buildEventDateFailDateEvent() throws Exception {
```

Comment:
reporting more than one error in the attempt to construct the EventDate object. Brave new world!